inc/driver-models/Gyroscope.h

```diff
@@ -0,0 +1,213 @@
+/*
+The MIT License (MIT)
+
+Copyright (c) 2017 Lancaster University.
+Copyright (c) 2018 Paul ADAM, Europe.
+
+Permission is hereby granted, free of charge, to any person obtaining a
+copy of this software and associated documentation files (the "Software"),
+to deal in the Software without restriction, including without limitation
+the rights to use, copy, modify, merge, publish, distribute, sublicense,
+and/or sell copies of the Software, and to permit persons to whom the
+Software is furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+DEALINGS IN THE SOFTWARE.
+*/
+
+#ifndef CODAL_GYROSCOPE_H
+#define CODAL_GYROSCOPE_H
+
+#include "CodalConfig.h"
+#include "CodalComponent.h"
+#include "Pin.h"
+#include "CoordinateSystem.h"
+#include "CodalUtil.h"
+
+/**
+  * Status flags
+  */
+#define GYROSCOPE_IMU_DATA_VALID               0x02
+
+/**
+  * Gyroscope events
+  */
+#define GYROSCOPE_EVT_DATA_UPDATE              1
+
+namespace codal
+{
+
+    /**
+     * Class definition for Gyroscope.
+     */
+    class Gyroscope : public CodalComponent
```

Comment:
Do you have any examples of drivers using this abstraction? (just for curiosity)

Reply:
https://github.com/LabAixBidouille-STM32/codal-stm32-iot-node/blob/master/model/STM32IotNodeGyroscope.h